### PR TITLE
Type ExternalLink and its dependencies

### PR DESCRIPTION
**Pull request by saramarcondes**
Types:

* Truncate
* Text
* Link
* ExternalLink
* Icon

### CODE REVIEW

File: packages/components/src/Text/types.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
+import { Props as TruncateProps } from '../Truncate/useTruncate';
```
Comment (saramarcondes) [inline packages/components/src/Text/types.ts:1]: @ItsJonQ This is what I was talking about using `.ts` files to define the types that we then import into the JSDoc 😉

File: packages/components/src/Text/useText.js [hunk 2]
```diff
@@ -128,7 +132,16 @@ export function useText(props) {
 	 */
 	if (!truncate && is.array(children)) {
 		content = React.Children.map(children, (child) => {
-			if (!is.plainObject(child)) return child;
+			// Thiis is... unfortunate. Would we rather throw a `ts-ignore` here than increase the complexity of this loop?
```
Comment (saramarcondes) [inline packages/components/src/Text/useText.js:135]: /cc @ItsJonQ @sirreal

File: packages/components/src/Text/types.ts (hunk 1 above)
Comment (ItsJonQ) [reply packages/components/src/Text/types.ts:1]: Thank goodness for that!~!!

File: packages/components/src/Text/useText.js (hunk 2 above)
Comment (ItsJonQ) [reply packages/components/src/Text/useText.js:135]: I mean. I'm okay with that 😂

**Comment by ItsJonQ** [comment]
@saramarcondes This is phenomenal!!! I feel like I'm able to type things given your examples - at least, at the `components` package level. (Goodness, I can't even at the `create-styles` or `context` level).

It's wonderful to see the dedicated `types.ts` file working 🙌 .
1. For the inline examples
2. Because it's easier to read/write (vs. JSDocs 😂 )

File: packages/components/src/Text/useText.js (hunk 2 above)
Comment (saramarcondes) [reply packages/components/src/Text/useText.js:135]: Yeah I think I'll just add some surgical ts-ignores. Otherwise this is too complex for a loop

**Pull request merged**